components/bl-gallery-component/src/index.js

```diff
@@ -0,0 +1,88 @@
+import { useMemo } from 'react';
+
+import { Gallery } from './components/gallery.js';
+
+const DefaultOptions = {
+  IMAGE_LABEL: 'Image',
+  SEPARATOR: 'of',
+  FADE_DURATION: 600,
+  IMAGE_FADE_DURATION: 600,
+  POSITION_FROM_TOP: 50,
+  RESIZE_DURATION: 700,
+};
+
+export default function GalleryComponent({ component }) {
+  const {
+    imagesData,
+    alwaysShowNavOnTouchDevices,
+    imageLabel,
+    separator,
+    disableScrolling,
+    fadeDuration,
+    imageFadeDuration,
+    positionFromTop,
+    resizeDuration,
+    showImageCount,
+    wrapAround,
+  } = component;
+
+  const options = useMemo(() => {
+    const {
+      IMAGE_LABEL,
+      SEPARATOR,
+      FADE_DURATION,
+      IMAGE_FADE_DURATION,
+      POSITION_FROM_TOP,
+      RESIZE_DURATION,
+    } = DefaultOptions;
+
+    return {
+      alwaysShowNavOnTouchDevices: alwaysShowNavOnTouchDevices === 'true',
+      imageLabel: imageLabel || IMAGE_LABEL,
+      separator: separator || SEPARATOR,
+      disableScrolling: disableScrolling === 'true',
+      fadeDuration: +fadeDuration >= 0 ? +fadeDuration : FADE_DURATION,
+      imageFadeDuration: +imageFadeDuration >= 0 ? +imageFadeDuration : IMAGE_FADE_DURATION,
+      positionFromTop: +positionFromTop >= 0 ? +positionFromTop : POSITION_FROM_TOP,
+      resizeDuration: +resizeDuration >= 0 ? +resizeDuration : RESIZE_DURATION,
+      showImageCount: showImageCount === 'true',
+      wrapAround: wrapAround === 'true',
```

Comment:
this approach is not good, because when changed at least one property it runs recalculation for all of the properties, much better to split the logic on several useMemo

Btw, if there is a simple calculation do not need to wrap it with useMemo, because, from my point of view, having a hook will be more complex for a program than comparing/calculating a simple operation like `wrapAround === 'true'`   or ` a + b`